server/sonar-server/src/main/java/org/sonar/server/computation/step/FeedPeriodsStep.java

```diff
@@ -0,0 +1,279 @@
+import org.apache.commons.lang.StringUtils;
+import org.sonar.api.CoreProperties;
+import org.sonar.api.config.Settings;
+import org.sonar.api.resources.Qualifiers;
+import org.sonar.api.utils.DateUtils;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.core.component.ComponentDto;
+import org.sonar.core.component.SnapshotDto;
+import org.sonar.core.component.db.SnapshotQuery;
+import org.sonar.core.persistence.DbSession;
+import org.sonar.server.computation.batch.BatchReportReader;
+import org.sonar.server.computation.component.Component;
+import org.sonar.server.computation.component.TreeRootHolder;
+import org.sonar.server.computation.period.Period;
+import org.sonar.server.computation.period.PeriodsHolderImpl;
+import org.sonar.server.db.DbClient;
+
+import static org.sonar.core.component.db.SnapshotQuery.SORT_FIELD.BY_DATE;
+import static org.sonar.core.component.db.SnapshotQuery.SORT_ORDER.ASC;
+import static org.sonar.core.component.db.SnapshotQuery.SORT_ORDER.DESC;
+
+/**
+ * Populates the {@link org.sonar.server.computation.period.PeriodsHolder}
+ *
+ * Here is how these periods are computed :
+ * - Read the 5 period properties ${@link CoreProperties#TIMEMACHINE_PERIOD_PREFIX}
+ * - Try to find the matching snapshots from the properties
+ * - If a snapshot is found, a new period is added to the repository
+ */
+public class FeedPeriodsStep implements ComputationStep {
+
+  private static final Logger LOG = Loggers.get(FeedPeriodsStep.class);
+
+  private static final int NUMBER_OF_PERIODS = 5;
+
+  private final DbClient dbClient;
+  private final Settings settings;
+  private final TreeRootHolder treeRootHolder;
+  private final BatchReportReader batchReportReader;
+  private final PeriodsHolderImpl periodsHolder;
+
+  public FeedPeriodsStep(DbClient dbClient, Settings settings, TreeRootHolder treeRootHolder, BatchReportReader batchReportReader,
+                         PeriodsHolderImpl periodsHolder) {
+    this.dbClient = dbClient;
+    this.settings = settings;
+    this.treeRootHolder = treeRootHolder;
+    this.batchReportReader = batchReportReader;
+    this.periodsHolder = periodsHolder;
+  }
+
+  @Override
+  public void execute() {
+    DbSession session = dbClient.openSession(false);
+    try {
+      periodsHolder.setPeriods(buildPeriods(session));
+    } finally {
+      session.close();
+    }
+  }
+
+  private List<Period> buildPeriods(DbSession session) {
+    Component project = treeRootHolder.getRoot();
+    ComponentDto projectDto = dbClient.componentDao().selectNullableByKey(session, project.getKey());
+    // No project on first analysis, no period
+    if (projectDto != null) {
+      List<Period> periods = new ArrayList<>(5);
+      PeriodResolver periodResolver = new PeriodResolver(session, projectDto.getId(), batchReportReader.readMetadata().getAnalysisDate(), project.getVersion(),
+        // TODO qualifier will be different for Views
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)